doc/whats_new/v1.1.rst

```diff
@@ -133,6 +133,12 @@ Changelog
   hole; when set to True, it returns the swiss-hole dataset. :pr:`21482` by
   :user:`Sebastian Pujalte <pujaltes>`.
 
+- |Enhancement| Introduce a new parameter `parser` in
+  :func:`datasets.fetch_openml`. This parameter allows to use `pandas.read_csv`
+  and its C parser to read ARFF file that is more memory efficient and much
+  faster. It will be used by default if `pandas` is installed.
+  :pr:`21938` by :user:`Guillaume Lemaitre <glemaitre>`.
```

Comment:
No, we do not do any magic that relies on whether pandas is installed or not. Since the 2 parsers are not 100% equivalent, this would cause silent bugs in user's call which is bad-bad-bad.

Import-magic is too dangerous and we already require pandas by default for `fetch_openml` because as `as_frame="auto"` is `True` for dense datasets in scikit-learn 1.0 (or even before).

```suggestion
- |Enhancement| Introduce the new parameter `parser="pandas"` in
  :func:`datasets.fetch_openml`. This parameter allows to use the very CPU
  and memory efficient `pandas.read_csv` parser to load dense ARFF
  formatted dataset files. If is possible to pass `parser="liac-arff"`
  to use the old LIAC parser.
  By default, dense datasets are loaded with "pandas" and sparse datasets
  are loaded with "liac-arff".
  :pr:`21938` by :user:`Guillaume Lemaitre <glemaitre>`.
```

We should probably add a note at the top level of the changelog to warn about the change of default parser for `fetch_openml`.